bloom-housing/bloom
# feat: application side nav

This PR addresses https://github.com/metrotranscom/doorway/issues/437

## Description

This PR utilizes the tab component so a user can filter the applications they would like to see. 

Note One: This implementation results in applications being re-fetched each time they select a new category. Based on the performance of this, I am wondering if there is a more reduced fetch call that only provides the required information.

Note Two: I made some assumptions on the mobile experience of this section, let me know what y'all think!

Note Three: I was looking into expanding the test coverage and was considering ctypress tests but know we have been cautious in adding more cypress. Additionally, seeing how limited my-applications.spec.ts is would it be worth it to expand it? Planning on expending integration test coverage in #4053 

## How Can This Be Tested/Reviewed?

This can be tested by signing into the public site and first testing the empty states of the different sections. If there are no applications at all, the message will be to browse more listings but if you have applied but none of the applications are in the selected state, it will be a custom message. Then apply to multiple listings. Lastly, change the status of those listings and observe the applications move to the corresponding section

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
